### ISSUE TITLE
想法不错，但是我感觉pm2完全能替代呢

### ISSUE THREAD

**Issue by simonNeo** [hua1995116/mmt#8]
没有其他意思，仅探讨。

**Comment by hua1995116** [comment]
pm2 主要是处理服务端守护进程，定位不一样，其实要说能代替，shell 什么都能干，只是麻烦。mmt 主要还是屏蔽复杂的命令抒写以及脚本管理。目前还支持了 package.json script 命令的快捷方式。例如package.json -> "dev": "xx webpack",  你只需要在当前目录运行 mmt add dev ，自动会注入路径以及执行命令。

**Comment by simonNeo** [comment]
定位来说确实不一样，但就使用逻辑上来说，两者使用方式差不多，pm2下是cd到项目目录，然后`pm2 start npm -- run dev/start`，下次在任何地方 `pm2 restart npm`（当然npm这个名字也是可以设置别名的）。很像对不对？

而且pm2作为知名的守护进程应用，学习了之后后续也有可能用得上，综上的话我还是比较推荐pm2。

当然mmt还是有有优势的，pm2作为守护进程当然会添加很多额外的功能，例如查看进程ID，历史日志等。这一点上mmt的优势在于：说明白了其实也就是个本地化数据读取分析+shell alias，所以都不用考虑内存/性能/xxx，连轻量都算不上😂
另外，我觉得mmt的另一个意义是，希望开发新手能体会到：原来我可以用这种方式提升自己的工作效率！这个点我觉得还是比较重要的。

ps：其实我日常处理你提到的这些场景，都是用Alfred完成的，这个墙裂推荐一下（不过我看你也有用到AppleScript，估计af也是基操了）。

**Comment by hua1995116** [comment]
是的。还有一点就是屏蔽了 AppleScript 控制 iTerm2 窗口的复杂命令，目前多条命令可以控制打开多个窗口。这个我觉得也是能为我提高生产力的一个巨大改变。

**Comment by hua1995116** [comment]
还有目前我手头前后端/mono repo项目比较多，所以现在开发重度依赖 mmt 去管理我的多个窗口
